### PR TITLE
Clarify draw_rectangle behaviour

### PR DESCRIPTION
**Pull request by dstansby**
The code in `draw_rectangle` was doing all kinds of acrobatics with long/lat and units, which is fine for a plot where longitude-latitude align with the x-y axes, but I have no idea what wild things would happen if they weren't. The code is much simpler with a call to `world_to_pixel`.

In addition, this means that the behavior for an arbitrary WCS is well defined, as now indicated in the docstring:
```
This draws a rectangle that has corners at ``(bottom_left, top_right)``,
and that has sides parallel to the x and y axes of the plot.
```

### CODE REVIEW

File: sunpy/map/mapbase.py [hunk 1]
```diff
@@ -1785,7 +1785,17 @@ def draw_limb(self, axes=None, **kwargs):
...
+
+        This draws a rectangle that has corners at ``(bottom_left, top_right)``,
+        and that has sides parallel to the x and y axes of the plot.
+
+        If ``width`` and ``height`` are specified, they are respectively added to the
+        longitude and latitude of the ``bottom_left`` coordinate to calculate a
+        ``top_right`` coordinate. This means that ``width`` and ``height`` will only
+        correspond to the width and height of the draw rectangle if the longitude
+        and latitude coordiante axes are parallel with the x and y axes of the pixels
+        that comprise the map.
```
Comment (dstansby) [inline sunpy/map/mapbase.py:1798]: @ayshih here is some text about what `width` and `height` mean. It feels a bit wordy to me, so suggestions/improvements very welcome.

**Comment by Cadair** [comment]
This isn't making assumptions about which axes is x and which is y, the "x" coordinate is just the first one in the WCS. WCSAxes handles transforming them to pixel coordinates as defined by the WCS.

**Comment by dstansby** [comment]
> This isn't making assumptions about which axes is x and which is y, the "x" coordinate is just the first one in the WCS.

The issue is the computation of `width` and `height`. Width is computed as
```
width = Longitude(top_right.spherical.lon - bottom_left.spherical.lon)
```
which is the difference in longitude between the coordinates, but when the rectangle is drawn this is used as the width along the x-axis. So if Longitude is on my y-axis, the `width` argument would currently specify the height of the rectangle as drawn.

> I don't really see any good reason to not use the transformation machinery in WCSAxes/matplotlib to do this.

I agree with this, but as far as I can tell the transformation machinery doesn't affect the `width` and `height` arguments to `plt.rectangle`, so these need to be computed manually anyway I think.

I'll try and come up with some simple examples to check the current behaviour of `draw_rectangle.`
